vendor/knative.dev/pkg/reconciler/events.go

```diff
@@ -60,13 +60,50 @@ func NewEvent(eventtype, reason, messageFmt string, args ...interface{}) Event {
 	}
 }
 
+// NewEventConditional returns an Event fully populated.
+func NewEventConditional(conditionFn ConditionalEventFn, eventtype, reason, messageFmt string, args ...interface{}) Event {
+	return &ReconcilerEvent{
+		EventType:   eventtype,
+		Reason:      reason,
+		Format:      messageFmt,
+		Args:        args,
+		ConditionFn: conditionFn,
+	}
+}
+
+// NewIfStatusUpdatedEvent returns an Event fully populated with the
+// conditional function set to watch for status updates from the leader.
+func NewEventIfStatusUpdated(eventtype, reason, messageFmt string, args ...interface{}) Event {
+	return &ReconcilerEvent{
+		EventType: eventtype,
+		Reason:    reason,
+		Format:    messageFmt,
+		Args:      args,
+		ConditionFn: func(s State) bool {
+			if s.IsStatusUpdated() && s.IsLeader() {
+				return true
+			}
+			return false
+		},
+	}
```

Comment:
this interdependency with `State` seems a bit messy - could the framework just elide `corev1.EventTypeNormal`?

Reply:
It can't. You don't know if that Normal event is "I did a reconcile" or "I fixed a thing externally"

Reply:
wouldn't `"I fixed a thing externally"` use the recorder directly?

Reply:
I guess my general point here is I'm not a fan of the `ConditionFn` - since it puts the onus on the event implementor to handle this correctly.

ie. I could forget about leader election

Reply:
The contract setup is that if a reconciler returns an event, it is logged. I was attempting to encourage single actions per reconciler loop with the returned event to avoid people from using the recorders directly.